docs/engineering-defaults.md

```diff
@@ -1,7 +1,147 @@
 # Engineering Defaults
 
+Much of OpenO11y's guidance is based on or enhanced by these best practices for
```

Comment:
We need to add more of the intro / definitions as to what/why Engineering Defaults.

Reply:
Let's iterate on this with additional PRs